### PR TITLE
cmd/snap-confine: fix exploding homedirs bug

### PR DESCRIPTION
**Pull request by zyga**
The combination of snap set system homedirs=/var/lib/ and a snap that causes a mimic to be created over /var/lib is sufficient to leak over to the initial mount namespace.

Fixes: https://warthogs.atlassian.net/browse/SNAPDENG-17066

### CODE REVIEW

File: cmd/snap-confine/mount-support.c [hunk 1]
```diff
@@ -927,7 +927,7 @@ static struct sc_mount *sc_homedir_mounts(const struct sc_invocation *inv)
 	for (int i = 0; i < inv->num_homedirs; i++) {
 		debug("Adding homedir: %s", inv->homedirs[i]);
 		mounts[i].path = sc_strdup(inv->homedirs[i]);
-		mounts[i].is_bidirectional = true;
+		mounts[i].is_bidirectional = false;
```
Comment (bboozzoo) [inline cmd/snap-confine/mount-support.c:930]:
```suggestion
		// Mounts done under homedir will not propagate to parent
		// mount namespace.
		mounts[i].is_bidirectional = false;
```

**Review by bboozzoo — APPROVED** [review]
LGTM

**Review by Meulengracht — APPROVED** [review]
Thank you for catching this!

**Comment by zyga** [comment]
I'm still exploring the corresponding area. I'll update this issue with an integration test.

**Review by alexmurray — APPROVED** [review]
LGTM - thanks for catching this @zyga (I am ashamed to admit this looks to have been due to a comment I made on the original PR where I appear to have been a bit confused about the impacts of bidirectional mounts)

**Comment by Meulengracht** [comment]
@zyga static checks are failing

**Comment by zyga** [comment]
@Meulengracht fixed

**Pull request merged**